platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -384,7 +391,10 @@ public void setStyleUrl(@NonNull String url) {
     if (destroyed) {
       return;
     }
-
+    if (nativeMapView == null) {
+      mapboxMapOptions.styleUrl(url);
```

Comment:
keeping state in mapboxMapOptions is map isn't ready yet, this will be applied to the map when started later on.